libgraphics/source/DescriptorAllocator/DescriptorLayoutCache.cxx

```diff
@@ -0,0 +1,70 @@
+#include "pivot/graphics/DescriptorAllocator/DescriptorLayoutCache.hxx"
+
+namespace pivot::graphics
+{
+
+bool DescriptorLayoutCache::DescriptorLayoutInfo::operator==(const DescriptorLayoutInfo &other) const
+{
+    if (other.bindings.size() != bindings.size()) {
+        return false;
+    } else {
+        for (unsigned i = 0; i < bindings.size(); i++) {
```

Comment:
Use [std::ranges::equal](https://en.cppreference.com/w/cpp/algorithm/ranges/equal) with a lambda converting to a tuple as the two projection functions. See the conversion to tuple below.